add_varsel_cmap.py

```diff
@@ -0,0 +1,158 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Modify an Emoji font to populate cmap 14 variation selector subtable.
+
+This uses the unicode variation selector data, finds all characters in the
+standard cmap that match, and generates a cmap format 14 subtable indicating
+that they have the provided presentation (emoji by default).  No fonts
+are processed if any already has a format 14 subtable.  Modified fonts
+are written to the provided dir and optionally have a provided suffix
+appended (before the extension).  An output file name can be provided
+in which case only one input file is allowed."""
+
+import argparse
+import os
+from os import path
+import re
+import sys
+
+from fontTools import ttLib
+from fontTools.ttLib.tables import _c_m_a_p as CMap
+
+from nototools import font_data
+
+UCD = path.join(path.dirname(__file__), 'third_party', 'ucd')
+
+def get_emoji_data():
+  """Parse emoji-data.txt and return a tuple of two sets of characters:
+  - those with a default emoji presentation
+  - those with a default text presentation"""
+
+  presentation_default_text = set()
+  presentation_default_emoji = set()
+  emoji_data = path.join(UCD, 'emoji-data.txt')
+  line_re = re.compile(r'([0-9A-F]{4,6})\s*;\s*(emoji|text)\s*;.*')
+  with open(emoji_data, 'r') as f:
+    for line in f.readlines():
+      m = line_re.match(line)
+      if m:
+        cp = int(m.group(1), 16)
+        if m.group(2) == 'emoji':
+          presentation_default_emoji.add(cp)
+        else:
+          presentation_default_text.add(cp)
+  return presentation_default_emoji, presentation_default_text
+
+
+def get_unicode_emoji_variants():
+  """Parse StandardizedVariants.txt and return a set of characters
+  that have a defined emoji variant presentation.  All such characters
+  also have a text variant presentation (as of v8.0.0), so a single
+  set works for both."""
+
+  emoji_variants = set()
+  variants_data = path.join(UCD, 'StandardizedVariants.txt')
+  line_re = re.compile(r'([0-9A-F]{4,6})\sFE0F;\s+emoji style;.*')
+  with open(variants_data, 'r') as f:
+    for line in f.readlines():
+      m = line_re.match(line)
+      if m:
+        emoji_variants.add(int(m.group(1), 16))
+  return emoji_variants
+
+
+def modify_font(font_name, font, presentation, emoji_variants):
+  cmap_table = font_data.get_cmap(font)
+  emoji = set(cmap_table.keys()) & emoji_variants
+  if not emoji:
+    print 'no emoji match those in %s' % font_name
+    return
+  uvs = 0xfe0f if presentation == 'emoji' else 0xfe0e
+  cmap14 = CMap.CmapSubtable.newSubtable(14)
+  cmap14.cmap = {}
+  cmap14.uvsDict = {uvs : [[c, None] for c in sorted(emoji)]}
+  cmap14.platformID = 0
+  cmap14.platEncID = 5
+  cmap14.language = 0xFF
```

Comment:
Some of these values are here because fontTools has a bug that prevents it from creating these values automatically.  0xFF is the value that fontTools normally would put here.  If it should be zero then that is another fontTools bug.